### DISCUSSION TITLE
Idea for a plugin: SPF resource record

### DISCUSSION THREAD

**Discussion by schmkr** [coredns/coredns#5867] [Ideas]
When dealing with email sending, one often has to lookup SPF records for a domain to determine if sending is allowed for that domain from the connecting ip address. 

The challenge with SPF records, is that more often than not, do not contain the actual ip address ranges that are allowed, but include directives to delegate the definition of the ip address ranges to the sending parties. For example, if you use Google Workspace on domain foo.bar, it requires an TXT record:

```
foo.bar.  300 IN TXT "v=spf1 include:_spf.google.com -all"
```

Then, you'd have to query `_spf.google.com` (which at the time of writing results in `include:_netblocks.google.com include:_netblocks2.google.com include:_netblocks3.google.com`) and you'd have to resolve those three includes as well. Until there are no include-directives left in all lookups and you only have a list of allowed ip ranges.

_What if_ a plugin in CoreDNS would allow for a (pseudo) resource record (e.g. `SPF`) that implements the logic of recursively performing the lookups and returning the end result of only the list of allowed ip ranges?

Is such a thing possible at all with a plugin?

**Comment by chrisohaver** [comment]
Yes a plugin could be written to do that.

I'm not sure how common it is for an email sender implementation to look up SPF records, but not know how to handle SPF includes.  If that's a common thing, then I think such a plugin would be quite useful.
